docs/environment-variables.md

```diff
@@ -52,6 +52,7 @@ This document outlines environment variables that can be used to customize behav
 | `WF_DEL_PROPAGATION_POLICY`              | `string`            | `""`                                                                                        | The deletion propagation policy for workflows.                                                                                                                                                                                                                           |
 | `WORKFLOW_GC_PERIOD`                     | `time.Duration`     | `5m`                                                                                        | The periodicity for GC of workflows.                                                                                                                                                                                                                                     |
 | `SEMAPHORE_NOTIFY_DELAY`                 | `time.Duration`     | `1s`                                                                                        | Tuning Delay when notifying semaphore waiters about availability in the semaphore                                                                                                                                                                                        |
+| `WATCH_CONTROLLER_CONFIGMAP`                | `bool`              | `true`                                                                                     | Whether controller configMap changes should cause the controller pod to restart.                                                                                                                                                                                                                              |
```

Comment:
This doesn't cause it to restart, it actually modifies values at run-time in memory. Also, as I've noted in #11855, this `RetryWatcher` is (currently) also responsible for semaphore ConfigMap updates as well, so it will disable that functionality as well.

As such we should reword this a good bit:
```suggestion
| `WATCH_CONFIGMAPS`                | `bool`              | `true`                                                                                     | Whether to watch the Controller's ConfigMap and semaphore ConfigMaps for run-time changes. When disabled, the Controller will only read ConfigMaps once and will have to be manually restarted to pick up new changes.                                                                                                  |
```